### ISSUE TITLE
How do I save it as a SVG format, when using p.on(ax).show() in seaborn 0.13

### ISSUE THREAD

**Issue by z626093820** [mwaskom/seaborn#3529]
![image](https://github.com/mwaskom/seaborn/assets/37371455/5b4d21eb-4696-4491-9ac8-ae2113f75f28)
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.load_dataset('diamonds')
sns.set('paper',style='ticks',font_scale=1.8)
plt.rcParams['font.sans-serif']='Arial'	#设置字体，必须放在sns.set后面。否则会被sns.set更新
plt.rcParams['svg.fonttype'] = 'none'	#保存svg时，字体为文本而不是路径,放在sns.set前面后面都可以
fig,ax=plt.subplots(figsize=(4,4))
sns.despine()

so.Plot(data, x="bill_length_mm", y="bill_depth_mm").layout(size=(3, 3))\
 .add(so.Dot(), color="species").label(x="a", y="b",title="c")\
 .add(so.Line(color="black"),so.PolyFit(), y="bill_depth_mm", label="depth").on(ax).show()
#p.on(ax).show()

THANK YOU

**Comment by mwaskom** [comment]
I’m not sure I understand the question. Showing and saving are distinct operations. Plot.show sends the plot to the pyplot (gui) frontend. There are different backends, but none if them really “save” a plot in any particular format.

**Comment by gambre09** [comment]
In this code, you create your Seaborn plot using the so module as you originally had. After creating the plot with p.on(ax).show(), you use plt.savefig to save the plot as an SVG file. Make sure to replace "your_plot_filename.svg" with your desired file path and name for the SVG file.

Code:-
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.load_dataset('diamonds')
sns.set('paper',style='ticks',font_scale=1.8)
plt.rcParams['font.sans-serif']='Arial' #设置字体，必须放在sns.set后面。否则会被sns.set更新
plt.rcParams['svg.fonttype'] = 'none' #保存svg时，字体为文本而不是路径,放在sns.set前面后面都可以
fig,ax=plt.subplots(figsize=(4,4))
sns.despine()

so.Plot(data, x="bill_length_mm", y="bill_depth_mm").layout(size=(3, 3))\
.add(so.Dot(), color="species").label(x="a", y="b",title="c")\
.add(so.Line(color="black"),so.PolyFit(), y="bill_depth_mm", label="depth")\
.on(ax).show()

# Save the plot as an SVG file
plt.savefig("your_plot_filename.svg", format='svg')

# Show the plot
plt.show()

# If you want to close the plot window after saving, you can add this line:
plt.close()

i was not able to get yout graph because the dataset was not given.
This is the example code where i have tried;

Code:-
import seaborn as sns
import matplotlib.pyplot as plt

# Create a sample Seaborn plot (replace this with your own plot code)
sns.set(style="whitegrid")
tips = sns.load_dataset("tips")
ax = sns.barplot(x="day", y="total_bill", data=tips)

# Save the plot as an SVG file
plt.savefig("your_plot_filename.svg", format='svg')

# Show the plot
plt.show()

# If you want to close the plot window after saving, you can add this line:
plt.close()

**Comment by mwaskom** [comment]
Are you following a tutorial or something? I can't speak for whether something is wrong in a third-party tutorial is incorrect and I'm finding the formatting of your messages very hard to follow. But if you want to save a Plot as svg, you can use [Plot.save](https://seaborn.pydata.org/generated/seaborn.objects.Plot.save.html).
